src/shared/backgroundScripts/listeners.js

```diff
@@ -20,18 +19,29 @@ export function initSharedListeners() {
       }
     });
 
+  // Update the toolbar icon when the tab is changed
+  browser.webNavigation.onCommitted.addListener(async (details) => {
```

Comment:
This event listener has no filter, and the event is therefore still very chatty.

Did you run into an issue when implementing the suggestion from: https://github.com/mozilla/firefox-launch/pull/18/files#r1468676561 ?

Reply:
Yes, in my attempts to disable or enable based on the filter, I could not catch some cases. For example, the about schema or new tabs in firefox, or chrome-extension pages in chrome. They would remain enabled when they should be disabled. 

I believe the issue lied with activeTab permission being enabled.

Reply:
In Chrome, you can use the declarativeContent API to achieve exactly the result you want, efficiently. Here is a code sample that you can use (add the declarativeContent permission to manifest.json):

https://github.com/Rob--W/crxviewer/blob/942b02fa871ff4ef6a29cf18b266711725314e86/src/background.js#L258-L331

In Firefox, that API is not available. But given the rarity of cases where the button should be disabled (and is not), I think that it would make more sense to accept the possibility of the button being in a wrong state (shown when it should be disabled) and handle that by showing a notification to the user when the extension doesn't accept the URL. Taking that approach would reduce the performance impact of the extension considerably, with minimal impact on the user experience.